src/Uno.Extensions.Navigation/ViewMapNotFoundException.cs

```diff
@@ -0,0 +1,141 @@
+﻿namespace Uno.Extensions.Navigation;
+
+/// <summary>
+/// Represents an exception that is thrown when a view map is not found in the view registry.
+/// </summary>
+public class ViewMapNotFoundException : Exception
```

Comment:
Hello @xperiandri, thanks for the PR! Maybe we could simplify a bit and have something like:

```csharp
public enum ViewMapSearchType
    View,
    ViewModel,
    Data,
    ResultData

public class ViewMapNotFoundException : Exception

    public ViewMapNotFoundException(ViewMapSearchType searchType, Type? type = null)
        : base(CreateNotFoundMessage(searchType, type))

    public ViewMapNotFoundException(ViewMapSearchType searchType, Type? type, Exception innerException)
        :base(CreateNotFoundMessage(searchType, type), innerException)

    private static string CreateNotFoundMessage(ViewMapSearchType searchType, Type? type)
```

Or if it's preferable to have multiple exception types (so they can be caught independently), we could define a base exception class like `ViewMapNotFoundExceptionBase`. In this base class, we can define the `CreateTypeNotFoundMessage` method such as:

```csharp
CreateTypeNotFoundMessage(string typeName, Type? type) 
    => $"The {typeName} type {type} was not found in the view registry.";
```

This method can then be used by the other exception classes that inherit from `ViewMapNotFoundExceptionBase`. Remember to use the `enum ViewMapSearchType`, please.

Reply:
I defined a base class.
Type comes from different errors, so different exception types needed